packages/utils/src/crypto/packet/keyDerivation.ts

```diff
@@ -0,0 +1,62 @@
+import { expand } from 'futoin-hkdf'
+import { PRIVATE_KEY_LENGTH, SECRET_LENGTH, HASH_ALGORITHM, HASH_LENGTH } from './constants'
+import { PRG_IV_LENGTH, PRG_KEY_LENGTH } from '../prg'
+import { PRP_IV_LENGTH, PRP_KEY_LENGTH } from '../prp'
+import type { PRGParameters } from '../prg'
+import type { PRPParameters } from '../prp'
+
+const HASH_KEY_BLINDING = 'HASH_KEY_BLINDING'
```

Comment:
What are these for?

Reply:
In modern crypto, you first extract the entropy from the given key material, `hkdf.extract`. Afterwards you create "virtual hash functions" for each key usage by using `hkdf.expand`, in our case: group element (un-)blinding, routing info (un-)blinding (PRG), and encryption / decryption (PRP). And to make sure that each of these "virtual hash functions" are indeed different, they get a different "hash key"

Reply:
so these are all inputs to hashes? Why not call them that?

Reply:
No, they aren't, these are hashkeys